zone/effects.cpp

```diff
@@ -678,6 +678,16 @@ void Client::ResetDisciplineTimer(uint32 timer_id) {
 	SendDisciplineTimer(timer_id, 0);
 }
 
+bool Client::HasDisciplineLearned(uint16 spell_id) {
+	bool has_learned = false;
+	for (auto index = 0; index < MAX_PP_DISCIPLINES; ++index) {
+		if (GetPP().disciplines.values[index] == spell_id) {
+			has_learned = true;
```

Comment:
We can just return true here and return false later.

Reply:
Figured one point of return was better than 2.

Reply:
well, if the first discipline checked is the one you're searching for, it's gonna still search the rest of them. Ideally you return early, you stop searching. Currently this always has the same complexity, where on average we should be hitting O(n/2).

Reply:
We could also just use std::find_if.

Reply:
I'll just add a break in the loop.